modules/javafx.graphics/src/main/java/com/sun/prism/d3d/D3DContext.java

```diff
@@ -141,13 +145,59 @@ private void setLost() {
         isLost = true;
     }
 
+    // KCR: debug
+    // KCR: simulate device removed
+    private int counter = 0;
+
+    private int simulateDeviceRemoved(int hr) {
+        if (hr != D3D_OK) return hr;
+
+        if (++counter >= 10) {
+            counter = 0;
+            System.err.println("simulate D3DERR_DEVICEREMOVED");
+            return D3DERR_DEVICEREMOVED;
+        }
+        return D3D_OK;
+    }
+
     /**
      * Validates the device, sets the context lost
      * status if necessary, and tries to restore the context if needed.
      */
     boolean testLostStateAndReset() {
+        if (disposed) return false;
+
         int hr = D3DResourceFactory.nTestCooperativeLevel(pContext);
 
+        // KCR: debug
+        // KCR: Periodically simulate device removed state
+//        hr = simulateDeviceRemoved(hr);
```

Comment:
If you want to test this on a Windows 10 system without using RDP, you can uncomment this line, which will cause a `D3DERR_DEVICEREMOVED` to be sent every 10 frames.